colosseum/declaration.py

```diff
@@ -533,14 +536,10 @@ def keys(self):
     ######################################################################
     def __str__(self):
         non_default = []
+        empty = '<EMPTY>'
         for name in _CSS_PROPERTIES:
-            try:
-                non_default.append((
-                    name.replace('_', '-'),
-                    getattr(self, '_%s' % name)
-                ))
-            except AttributeError:
-                pass
+            if getattr(self, '_%s' % name, empty) != empty:
+                non_default.append((name.replace('_', '-'), getattr(self, name)))
```

Comment:
Ok!

Reply:
We will need something like this for shorthand properties, since the value is constructed by the property `name` and not `_name`

```python
    def __str__(self):
        non_default = []
        for name in _CSS_PROPERTIES:
            try:
                getattr(self, '_%s' % name)
                non_default.append((name.replace('_', '-'), getattr(self, name)))
            except AttributeError:
                pass

        return "; ".join(
            "%s: %s" % (name, value)
            for name, value in sorted(non_default)
        )
```

Reply:
Fixed?

Reply:
Yes - although I'm still a little confused on the condition where `getattr(self, name)` will succeed but `getattr(self, '_%s' % name)` will fail...

Reply:
Fixed! (no longer needed)